tests/lib/rules/prefer-destructuring.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @fileoverview Prefer destructuring from arrays and objects
+ * @author Alex LaFroscia
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/prefer-destructuring"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
```

Comment:
Nit: Instead of repeating `{ parserOptions: 6 }` for each test, it would be better to pass this as a default to the `RuleTester` constructor.

```js
const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 6 } });
```